### PR TITLE
Update cmake-embedded.yml

### PR DESCRIPTION
**Pull request by xbenng**

### CODE REVIEW

File: .github/workflows/cmake-embedded.yml [hunk 1]
```diff
@@ -89,7 +89,7 @@ jobs:
         tag: ${{steps.gitversion.outputs.major}}.${{steps.gitversion.outputs.minor}}.${{steps.gitversion.outputs.patch}}+${{steps.gitversion.outputs.build}}-${{ steps.branch-name.outputs.current_branch }}
 
     - name: Delete dev Branch
-      if: ${{ startsWith(github.ref, 'refs/heads/dev') }}
+      if: startsWith(github.ref, 'refs/heads/dev')
       run: |
         git checkout -b temp \
```
Comment (willson556) [inline .github/workflows/cmake-embedded.yml]: I don't think you need this checkout step

Comment (xbenng) [reply .github/workflows/cmake-embedded.yml]: How do you suggest getting branch name?

Comment (willson556) [reply .github/workflows/cmake-embedded.yml]: Does:
```yaml
    - name: Get branch name
       id: branch-name
       uses: tj-actions/branch-names@v5.1
```
not work?

Comment (xbenng) [reply .github/workflows/cmake-embedded.yml]: oh nevermind. the GitHub app was displaying this comment on the wrong line.

**Comment by harmandersihra** [comment]
What is this for?

**Comment by xbenng** [comment]
> What is this for?

this is to enable releasing from a development branch

**Pull request merged**